apiserver/src/main/java/org/dependencytrack/resources/v2/AdvisoriesResource.java

```diff
@@ -0,0 +1,365 @@
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import org.dependencytrack.persistence.jdbi.ProjectDao;
+import org.glassfish.jersey.media.multipart.FormDataContentDisposition;
+import org.glassfish.jersey.media.multipart.FormDataParam;
+import io.csaf.retrieval.RetrievedDocument;
+import org.dependencytrack.datasource.vuln.csaf.CsafSource;
+import org.dependencytrack.datasource.vuln.csaf.ModelConverter;
+import org.dependencytrack.parser.dependencytrack.BovModelConverter;
+import org.dependencytrack.model.VulnerableSoftware;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.time.Instant;
+
+import org.dependencytrack.auth.Permissions;
+import org.dependencytrack.model.Advisory;
+import org.dependencytrack.model.Vulnerability;
+import org.dependencytrack.persistence.QueryManager;
+import org.dependencytrack.persistence.jdbi.AdvisoryDao;
+import org.dependencytrack.resources.AbstractApiResource;
+import org.dependencytrack.resources.v1.openapi.PaginatedApi;
+import org.dependencytrack.resources.v1.problems.ProblemDetails;
+
+import java.util.List;
+import java.util.UUID;
+
+import static org.dependencytrack.persistence.jdbi.JdbiFactory.inJdbiTransaction;
+import static org.dependencytrack.persistence.jdbi.JdbiFactory.withJdbiHandle;
+import static org.dependencytrack.resources.v1.FindingResource.MEDIA_TYPE_SARIF_JSON;
+
+/**
+ * JAX-RS resources for advisories.
+ *
+ * @author Lawrence Dean
+ * @since 5.7.0
+ */
+@Path("/advisories")
+@Tag(name = "advisories")
+@SecurityRequirements({
+        @SecurityRequirement(name = "ApiKeyAuth"),
+        @SecurityRequirement(name = "BearerAuth")
+})
+public class AdvisoriesResource extends AbstractApiResource {
+    private static final Logger LOGGER = Logger.getLogger(AdvisoriesResource.class);
+
+    @GET
+    @Produces({ MediaType.APPLICATION_JSON, MEDIA_TYPE_SARIF_JSON })
+    @Operation(summary = "Returns a list of all advisories", description = "<p>Requires permission <strong>VIEW_VULNERABILITY</strong></p>")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "A list of all advisories for a specific project, or a SARIF file", headers = @Header(name = TOTAL_COUNT_HEADER, description = "The total number of advisories", schema = @Schema(format = "integer")), content = {
+                    @Content(array = @ArraySchema(schema = @Schema(implementation = AdvisoryDao.AdvisoryInProjectRow.class)), mediaType = MediaType.APPLICATION_JSON),
+                    @Content(schema = @Schema(type = "string"), mediaType = MEDIA_TYPE_SARIF_JSON)
+            }),
+            @ApiResponse(responseCode = "401", description = "Unauthorized"),
+            @ApiResponse(responseCode = "403", description = "Access to advisories prohibited", content = @Content(schema = @Schema(implementation = ProblemDetails.class), mediaType = ProblemDetails.MEDIA_TYPE_JSON)),
+    })
```

Comment:
Does not return SARIF.

```suggestion
    @GET
    @Produces(MediaType.APPLICATION_JSON)
    @Operation(summary = "Returns a list of all advisories", description = "<p>Requires permission <strong>VIEW_VULNERABILITY</strong></p>")
    @ApiResponses(value = {
            @ApiResponse(responseCode = "200", description = "A list of all advisories for a specific project", headers = @Header(name = TOTAL_COUNT_HEADER, description = "The total number of advisories", schema = @Schema(format = "integer")), content = @Content(array = @ArraySchema(schema = @Schema(implementation = AdvisoryDao.AdvisoryInProjectRow.class)), mediaType = MediaType.APPLICATION_JSON)),
            @ApiResponse(responseCode = "401", description = "Unauthorized"),
            @ApiResponse(responseCode = "403", description = "Access to advisories prohibited", content = @Content(schema = @Schema(implementation = ProblemDetails.class), mediaType = ProblemDetails.MEDIA_TYPE_JSON)),
    })
```